### DISCUSSION TITLE
\u0000导致告警失败

### DISCUSSION THREAD

**Discussion by vicStudio** [clickvisual/clickvisual#1037] [Q&A]
查看clickvisual.log日志文件，filterId等字段多了很多\u0000字符导致告警失败，日志内容如下：
`{"lv":"info","ts":1692240748,"msg":"alarm","notification":{"version":"4","groupKey":"{}:{alertname=\"ClickVisual-8f72ac67_4b97_4089_bec1_7d6e0655bacd_20\"}","status":"firing","receiver":"feishu\\.web\\.hook","groupLabels":{"alertname":"ClickVisual-8f72ac67_4b97_4089_bec1_7d6e0655bacd_20"},"commonLabels":{"alarmId":"1\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","alertname":"ClickVisual-8f72ac67_4b97_4089_bec1_7d6e0655bacd_20","filterId":"20\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","severity":"warning","uuid":"8f72ac67-4b97-4089-bec1-7d6e0655bacd\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"},"commonAnnotations":{"description":"  (当前值: 2)","summary":"告警 "},"externalURL":"http://05c1d874fbeb:9093","alerts":[{"labels":{"alarmId":"1\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","alertname":"ClickVisual-8f72ac67_4b97_4089_bec1_7d6e0655bacd_20","filterId":"20\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","severity":"warning","uuid":"8f72ac67-4b97-4089-bec1-7d6e0655bacd\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"},"annotations":{"description":"  (当前值: 2)","summary":"告警 "},"startsAt":"2023-08-17T02:51:13.257Z","endsAt":"0001-01-01T00:00:00Z"}]}}`

开发环境：
clickvisual版本：master分支（73b42ab4d41924398cab63f465349d687f5e76fd）
clickhouse版本：22.10
prometheus版本：2.37.5
alertmanager版本：0.25.0

请问如何排错？

**Comment by kl7sn** [comment]
这个是 alertmanager 回传消息到 clickvisual 的日志打印，暂时没有排查出原因，我试试是否可以过滤。

**Comment by chenmonster** [comment]
我是 alertmanager 显示的就是乱码，然后查询 clickvisual 容器的日志，看到回调也是乱码的，没找到解决方法
```json
{
    "lv": "info",
    "ts": 1726142144,
    "msg": "alarm",
    "lname": "clickvisual.log",
    "notification": {
        "version": "4",
        "groupKey": "{}:{alertname=\"ClickVisual-30559d04_e00f_4bbe_a14b_e25f0b3b4d56_9\"}",
        "status": "resolved",
        "receiver": "webhook",
        "groupLabels": {
            "alertname": "ClickVisual-30559d04_e00f_4bbe_a14b_e25f0b3b4d56_9"
        },
        "commonLabels": {
            "alarmId": "4\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
            "alertname": "ClickVisual-30559d04_e00f_4bbe_a14b_e25f0b3b4d56_9",
            "filterId": "9\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
            "severity": "warning",
            "uuid": "30559d04-e00f-4bbe-a14b-e25f0b3b4d56\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"
        },
        "commonAnnotations": {
            "description": "  (当前值: 1)",
            "summary": "告警 "
        },
        "externalURL": "http://53ee17ba4f94:9093",
        "alerts": [
            {
                "labels": {
                    "alarmId": "4\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
                    "alertname": "ClickVisual-30559d04_e00f_4bbe_a14b_e25f0b3b4d56_9",
                    "filterId": "9\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
                    "severity": "warning",
                    "uuid": "30559d04-e00f-4bbe-a14b-e25f0b3b4d56\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"
                },
                "annotations": {
                    "description": "  (当前值: 1)",
                    "summary": "告警 "
                },
                "startsAt": "2024-09-12T11:50:14.362Z",
                "endsAt": "2024-09-12T11:51:29.362Z"
            }
        ]
    }
}
```
